synapse/util/caches/lrucache.py

```diff
@@ -88,23 +92,41 @@ def delete_node(node):
             prev_node.next_node = next_node
             next_node.prev_node = prev_node
 
+            for cb in node.callbacks:
+                cb()
+            node.callbacks = []
+
         @synchronized
-        def cache_get(key, default=None):
+        def cache_get(key, default=None, callback=None):
             node = cache.get(key, None)
             if node is not None:
                 move_node_to_front(node)
+                if callback:
+                    node.callbacks.append(callback)
```

Comment:
If the cache entry doesn't get exipired is there a risk that many callbacks could accumulate here over time?